createFeatureFlag/main.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/Real-Dev-Squad/feature-flag-backend/database"
+	"github.com/Real-Dev-Squad/feature-flag-backend/models"
+	"github.com/Real-Dev-Squad/feature-flag-backend/utils"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/go-playground/validator/v10"
+	"github.com/google/uuid"
+	"log"
+	"net/http"
+	"time"
+)
+
+var validate *validator.Validate
+
+func init() {
+	validate = validator.New()
+}
+
+func createFeatureFlag(db *dynamodb.DynamoDB, createFeatureFlagRequest models.CreateFeatureFlagRequest) error {
+	featureFlag := models.FeatureFlag{
+		Id:          uuid.New().String(),
+		Name:        createFeatureFlagRequest.FlagName,
+		Description: createFeatureFlagRequest.Description,
+		CreatedBy:   createFeatureFlagRequest.UserId,
+		UpdatedBy:   createFeatureFlagRequest.UserId,
+		CreatedAt:   time.Now().Unix(),
+		UpdatedAt:   time.Now().Unix(),
+		Status:      utils.ENABLED,
+	}
+
+	//marshal to ddb attribute value
+	item, err := dynamodbattribute.MarshalMap(featureFlag)
+	if err != nil {
+		log.Println("Error marshalling object to DynamoDB AttributeValue: %v", err)
+		return err
+	}
+
+	log.Println("item is", item)
+	input := &dynamodb.PutItemInput{
+		TableName: aws.String(database.GetFeatureFlagTableName()),
+		Item:      item,
+	}
+
+	log.Println(db, " is the db object")
+	_, err = db.PutItem(input)
```

Comment:
This line is trying to save the feature flag to DynamoDB. However, the _ variable is not being used. This means that the error returned by db.PutItem() is not being captured.

Reply:
https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.44.266/service/dynamodb#DynamoDB.PutItem If we see the `putItem` output its of the form `*PutItemOutput, err`, we are capturing the err coming in err var.
Let me know if I'm understanding it incorrectly @iamitprakash